flybywiresim/a32nx
# fix(efb): simbridge remote setting fix

Fixes #7933

## Summary of Changes

Added extra logic to the getSimBridgeUrl() function to default to 'localhost' for the IP address when the SimBridge is in local mode rather than remote. This way, a user switching between 'This PC' and 'Remote PC' under the FlyPad Sim Options doesn't need to manually change the IP to 'localhost' for local SimBridge to work and can retain the saved IP for switching back.

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions


- Load into Aircraft on runway
- Start SimBridge server
- Observe that SimBridge connects automatically (assuming correct settings)
- Open Remote MCDU or any other means of testing SimBridge connection
- Go to FlyPad -> Settings -> Sim Options -> Set 'SimBridge Host Machine' to 'Remote PC' and enter a bogus IP address below
- Observe that SimBridge connection disconnects (Remote MCDU stops working)
- Go to FlyPad -> Settings -> Sim Options -> Set 'SimBridge Host Machine' to 'This PC'
- Observe that SimBridge reconnects as intended (Remote MCDU starts working again)
- Go to FlyPad -> Settings -> Sim Options -> Set 'SimBridge Host Machine' to 'Remote PC' and observe the IP address is retained 


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
